testsuite/elytron/src/test/java/org/wildfly/test/security/common/elytron/SecurityDomainTest.java

```diff
@@ -0,0 +1,45 @@
+package org.wildfly.test.security.common.elytron;
+
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+import static org.hamcrest.CoreMatchers.containsString;
+
+@RunWith(WildflyTestRunner.class)
+public class SecurityDomainTest {
+    CLIWrapper cli;
+
+    @Before
+    public void setup() throws Exception {
+        cli = new CLIWrapper(true);
+        // add security-domain
+        cli.sendLine("/subsystem=elytron/filesystem-realm=myFsRealm:add(path=my-fs-realm-users,relative-to=jboss.server.config.dir)");
+        cli.sendLine("/subsystem=elytron/filesystem-realm=myFsRealm:add-identity(identity=myIdentity)");
+        cli.sendLine("/subsystem=elytron/security-domain=mySD:add(realms=[{realm=myFsRealm}],default-realm=myFsRealm)");
+        cli.sendLine("reload");
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        removeTestResources();
+        cli.close();
+    }
+
+    private void removeTestResources() {
+        cli.sendLine("/subsystem=elytron/security-domain=mySD:remove");
+        cli.sendLine("/subsystem=elytron/filesystem-realm=myFsRealm:remove");
+        cli.sendLine("reload");
+    }
+
+    @Test
+    public void testValidHostContextMapValue() {
+        boolean success = cli.sendLine("/subsystem=elytron/security-domain=mySD:read-identity(name=myIdentity)", true);
+        Assert.assertFalse(success);
+        Assert.assertThat("Wrong error message", cli.readOutput(), containsString("Identity with name [myIdentity] not authorized."));
```

Comment:
@darranl @fjuma After fix for WFCORE-4407 security domain service was not started before attempting to read the identity which resulted in NPE. 

This fix removes NPE but it is still not possible to read the identity and resulting message is "Identity with name [myIdentity] not authorized." This is also invalid response as I should be able to read the identity I just added, right? If so, I will create separate issue for this.

Reply:
Just a clarification that the response `Identity with name [myIdentity] not authorized.` is unrelated to this NPE bug. This NPE bug is happening since wf-core 9.0.0.Beta7 and before that the response was also `Identity with name [myIdentity] not authorized.`

Reply:
update: when I add `default-permission-mapper` to the security domain I can read the identity. I have updated the PR